server/src/main/scala/com/cloudera/livy/server/interactive/CreateInteractiveRequest.scala

```diff
@@ -36,5 +36,5 @@ class CreateInteractiveRequest {
   var queue: Option[String] = None
   var name: Option[String] = None
   var conf: Map[String, String] = Map()
-
+  var addHiveDeps: Boolean = false
```

Comment:
This is touching a public API... while adding fields is not bad, I'm not sure this is the right way to do it.

The right way would be to read the Hive conf and see whether the datanucleus jars are needed (which is basically what this option translates to). Forcing users to do this is not very user-friendly.